executor-scripts/linux/vrrp

```diff
@@ -0,0 +1,347 @@
+#!/bin/sh
+# This executor is responsible for setting up the Virtual Router Redundancy Protocol (VRRP) overlay interfaces.
+#
+# Copyright (C) 2022 EasyNetDev <devel@easynet.dev>
+#
+# Permission to use, copy, modify, and/or distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# This software is provided 'as is' and without any warranty, express or
+# implied. In no event shall the authors be liable for any damages arising
+# from the use of this software.
+#
+# Sun, 21 Apr 2023 10:36:31 +0300
+#  -- EasyNetDev <devel@easynet.dev>
+#
+# This executor will do the configuration for MACVLAN virtual interfaces.
+#
+# Each MACVLAN interface must be controlled by a VRRP daemon, like FRRouting.
+#
+# Options needed for VRRP executro to be able to create the MACVLAN interface are:
+#
+# IFACE                Main interface name
+# IF_VRRP_ID           VRID list to create the interfaces only
+# IF_VRRP_ADDRESS      IP list to create the interfaces and add IPs (Optional)
+#
+# Interface naming would be in format vXXXXXXXX-YY-V
+# Where
+#  ->  XXXXXXXX is CRC32 value calculated for main interface name
+#  ->  YY	is HEX value of the VRRP ID
+#  ->  V	is IP version 4 or 6
+#
+# This keeps the a fix VRRP interface name length to 14 characters (maximum are 15 characters in Linux)
+# and the part vXXXXXXXX-YY is the same for both IP protocols.
+# Also it keeps the same names after reboots if the main interface name is not changed or the CRC32 tool is not changed.
+#
+# This script is written in shell and is not using any Bash built-in functions.
+#
+
+[ -n "$VERBOSE" ] && set -x
+
+# Global variables for CRC32 tool
+crc32_cmd=""
+crc32_tool=""
+
+################################################################################
+#                        VRRP CRC32 tool functions                             #
+################################################################################
+
+# In case the crc32_tool is in auto, search for crc32b then for cksum
+find_crc32_tools_in_auto() {
+    # Search for crc32b
+    # In case we are in test mode, check for current build directory for crc32b
+    if [ -n "${MOCK}" -a -z ${crc32_cmd} ]; then
+	crc32_tool="MOCK"
+	return
+    fi
+    crc32_cmd=$(which crc32sum)
+    if [ -n "${crc32_cmd}" ]; then
+	crc32_tool="crc32sum"
+	return
+    fi
+    # Search for cksum
+    crc32_cmd=$(which cksum)
```

Comment:
As above.

Reply:
All crc32_cmd I've changed them to :
```
    crc32_cmd=$(command -v crc32sum)
```
or
```
    crc32_cmd=$(command -v cksum)
```